### PR TITLE
Custom state: Preparation for TAG review

### PR DESCRIPTION
**Pull request by tkent-google**
Add "Security and Privacy Considerations", "Major open issues", "Discussions", and "Self-Review Questionnare on Security and Privacy" sections

### CODE REVIEW

**Comment by tkent-google** [comment]
@domenic @rakina would you review this please?

File: proposals/custom-states-and-state-pseudo-class.md [hunk 1]
```diff
@@ -120,3 +120,91 @@ This causes compatibility issues if we ever want to introduce new CSS pseudo-cla
...
+
+No differences.  This feature should work regardless of first-party or third-party contexts.
+
+### 2.14. How does this specification work in the context of a user agent’s Private Browsing or "incognito" mode?
+
+No differences.  This features should work in such mode, and sites can't detect such mode with this feature.
+
+### 2.15. Does this specification have a "Security Considerations" and "Privacy Considerations" section?
+
+Yes.
```
Comment (annevk) [inline proposals/custom-states-and-state-pseudo-class.md:202]: Nothing relevant in there for this feature though, right?

File: proposals/custom-states-and-state-pseudo-class.md [hunk 2]
```diff
@@ -120,3 +120,91 @@ This causes compatibility issues if we ever want to introduce new CSS pseudo-cla
...
     
 We are open to other suggestions. However the discussion so far seems to favor `:state(foo)`.
+
+## Security and Privacy Considerations
+
+This feature must not have security and privacy implications.  The feature works within a single document, and nothing is exposed to outside of the document.
+
+## Major open issues
+
+-   Web components should be able to expose a list of supported states to web components users.
```
Comment (rakina) [inline proposals/custom-states-and-state-pseudo-class.md]: I don't think this is a major issue, IIRC something like this was discussed in TPAC and we don't really need to provide a way to solve this

File: proposals/custom-states-and-state-pseudo-class.md (hunk 1 above)
Comment (tkent-google) [reply proposals/custom-states-and-state-pseudo-class.md:202]: Right.
I think we need to provide "Self-Review Questionnare on Security and Privacy" anyway according to the TAG review request template.

File: proposals/custom-states-and-state-pseudo-class.md (hunk 2 above)
Comment (tkent-google) [reply proposals/custom-states-and-state-pseudo-class.md]: Ok, removed it.

File: proposals/custom-states-and-state-pseudo-class.md (hunk 1 above)
Comment (annevk) [reply proposals/custom-states-and-state-pseudo-class.md:202]: Yeah, I guess I'd lean towards answering no here. (Or "Yes, but no need to cover this feature.")

Comment (tkent-google) [reply proposals/custom-states-and-state-pseudo-class.md:202]: I think https://www.w3.org/TR/security-privacy-questionnaire/#considerations recommends to add the section even if the feature has no implication.  So this PR adds the "Security and Privacy Considerations" section.

**Review by domenic — APPROVED** [review]
Sorry that I missed this.

**Pull request merged**